File: Dec2020/28Dec/DefuseTheBomb.java
/*
Problem Name: 1652. Defuse the Bomb
Problem Link: https://leetcode.com/problems/defuse-the-bomb/
*/
class Solution {
    public int[] decrypt(int[] code, int k) {
        int key[] = new int[code.length];
        int n = code.length;
        for(int i=0;i<n;i++){
            int sum = 0, d;
            if(k < 0)
                d = -1;
            else
                d = 1;
            for(int j=(i+d+n)%n,cnt=0;cnt<Math.abs(k);j=(j+d+n)%n,cnt++)
                sum += code[j];
            key[i] = sum;
        }
        return key;
    }
}

File: Dec2020/28Dec/DefuseTheBomb.cpp
/*
Problem name: 1652. Defuse the Bomb
Problem link: https://leetcode.com/problems/defuse-the-bomb/
*/

class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {

        vector<int>copy = code;
        int sum, n = code.size();
        
        if(k<0){
            k=-1*k;
            for(int i=0;i<n;++i){
                sum=0;
                for(int j=1;j<=k;++j){
                    if(i-j<0)
                        sum += copy[n+(i-j)];
                    else
                        sum += copy[i-j];
                }
                code[i]=sum;
            }
        }
        
        else if(k>0){
            for(int i=0;i<n;++i){
                sum=0;
                for(int j=1;j<=k;++j){
                    sum += copy[(i+j)%n];
                }
                code[i]=sum;
            }
        }
        
        else{
            for(int i=0;i<n;++i){
                code[i]=0;
            }
        }
        return code;
    }
};
